crates/llms/src/chat/mistral.rs

```diff
@@ -407,141 +443,96 @@ impl Chat for MistralLlama {
             )
-            .await?;
+            .await
+            .boxed()
+            .context(FailedToRunModelSnafu)?;
 
         match resp.choices.first() {
             Some(choice) => Ok(choice.message.content.clone()),
             None => Ok(None),
         }
     }
 
-    #[allow(deprecated, clippy::cast_possible_truncation)]
+    async fn chat_stream(
+        &self,
+        req: CreateChatCompletionRequest,
+    ) -> Result<ChatCompletionResponseStream, OpenAIError> {
+        let recver = self.send_message(req).await.map_err(|e| {
+            OpenAIError::ApiError(ApiError {
+                message: e.to_string(),
+                r#type: None,
+                param: None,
+                code: None,
+            })
+        })?;
+        Ok(stream_from_response(recver))
+    }
+
     async fn chat_request(
         &self,
         req: CreateChatCompletionRequest,
     ) -> Result<CreateChatCompletionResponse, OpenAIError> {
-        let messages = RequestMessage::Chat(
-            req.messages
-                .iter()
-                .map(message_to_mistral)
-                .collect::<Vec<_>>(),
-        );
-
-        let tools: Option<Vec<Tool>> = req.tools.map(|t| t.iter().map(convert_tool).collect());
-        let tool_choice: Option<ToolChoice> = req.tool_choice.map(|s| convert_tool_choice(&s));
-
-        // TODO confirm this transformtion.
-        let p = SamplingParams {
-            temperature: req.temperature.map(f64::from),
-            top_k: None,
-            top_p: req.top_p.map(f64::from),
-            min_p: None,
-            top_n_logprobs: req.top_logprobs.unwrap_or_default().into(),
-            frequency_penalty: req.frequency_penalty,
-            presence_penalty: req.presence_penalty,
-            stop_toks: req.stop.map(|s| match s {
-                Stop::String(s) => mistralrs::StopTokens::Seqs(vec![s]),
-                Stop::StringArray(s) => mistralrs::StopTokens::Seqs(s),
-            }),
-            max_len: req.max_completion_tokens.map(|x| x as usize),
-            // logits_bias: req.logit_bias,
-            logits_bias: None,
-            n_choices: req.n.unwrap_or(1) as usize,
-            dry_params: None,
+        let mut recver = self.send_message(req).await.map_err(|e| {
+            OpenAIError::ApiError(ApiError {
+                message: e.to_string(),
+                r#type: None,
+                param: None,
+                code: None,
+            })
+        })?;
+        let Some(resp) = recver.recv().await else {
+            return Err(OpenAIError::ApiError(ApiError {
+                message: "model pipeline unexpectedly closed".to_string(),
+                r#type: None,
+                param: None,
+                code: None,
+            }));
         };
 
-        let resp = self
-            .run_internal(messages, tools, tool_choice, Some(p))
-            .await
+        let z = Self::post_process_req(resp)
```

Comment:
Naming: `z` is hard to read/understand the purpose, please see if there could be a better alternative